src/detections/rule/matchers.rs

```diff
@@ -502,10 +497,26 @@ impl LeafMatcher for DefaultMatcher {
     }
 
     fn is_match(&self, event_value: Option<&String>, recinfo: &EvtxRecordInfo) -> bool {
-        // PipeElement::EqualsFieldが設定されていた場合
-        let pipe = self.pipes.first().unwrap_or(&PipeElement::Wildcard);
-        if pipe.get_eqfield().is_some() {
-            return pipe.is_eqfield_match(event_value, recinfo);
+        let pipe: &PipeElement = self.pipes.first().unwrap_or(&PipeElement::Wildcard);
+        let match_result = match pipe {
+            PipeElement::EqualsField(_) | PipeElement::Endswithfield(_) => {
+                Some(pipe.is_eqfield_match(event_value, recinfo))
+            }
+            PipeElement::Cidr(ip_result) => match ip_result {
+                Ok(matcher_ip) => {
+                    let event_value_str = event_value.unwrap();
```

Comment:
It crashes here when the IpAddress field does not exist, so you might want to modify it as follows

```suggestion
                    let event_value_str = event_value.unwrap_or_default();
```

Reply:
This fix should solve the problem with https://github.com/Yamato-Security/hayabusa/pull/961#issuecomment-1470976731

Reply:
Thank you so much for looking into the cause of the problem! :)